frontend/src/pages/projects/update/[user]/[projectName].js

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useState, useContext } from 'react'
+import { useRouter } from 'next/router'
+
+import { Page } from '@components/Page'
+import DropZone from '@components/DropZone'
+import FilesPreview from '@components/FilesPreview'
+import useForm from '@hooks/useForm'
+import { ProjectUpdateForm } from '@models/ProjectUpdateForm'
+import { UploadContext } from '@contexts/UploadContext'
+import { getRepo, updateRepo } from '@utils/giteaApi'
+import { Button, Form, Header, Input, Segment, TextArea } from 'semantic-ui-react'
+
+const UpdateProject = () => {
+  const router = useRouter()
+  const { user, projectName, create } = router.query
+  const { setPersistenceScope } = useContext(UploadContext)
+  const [project, setProject] = useState({})
+
+  const fullname = `${user}/${projectName}`
+
+  useEffect(() => {
+    const getProject = async () => {
+      const project = await getRepo(fullname)
+      setProject(project)
+    }
+    setPersistenceScope(projectName)
+    getProject().then()
+  }, [])
```

Comment:
Everywhere were you have an empty `.then()` something strange is going on.

1. An empty `.then()` is a no-op, not sure why you have added these at all
2. Most of the time your use-effects can be simplified and made clearer by avoiding async/await, e.g this one could be:

```js
  useEffect(() => {
    setPersistenceScope(projectName)
    getRepo(fullname).then(setProject)
  }, [])
```

Reply:
You still need to do a search and delete any remaining empty `.then()`.